content/iOS/Practices/iOS-practices.md

```diff
@@ -0,0 +1,116 @@
+# A few iOS practices that can go a long way when maintaining your projects.
+
+## Introduction 
+
+Have you ever had to maintain or add features to a project you developed in the past and think: who wrote this and why do they hate me so much? 
+
+I think this is quite normal if we have in mind that some projects grow way bigger than we initially expected, and even when you have the best intentions, and you put lots of love and hard work in them, as you mature as a developer you start incorporating more and more practices, and the way you develop changes drastically, making it difficult and annoying to go back to older projects.
+
+I wanted to write this article to help me and others avoid this feeling when retaking an old project.
+
+The following are a few practices I've been incorporating and think it will help keep my next projects clean, easily maintainable, and extensible. 
+
+## Project Structure
+
+This may be one of the more obvious points for seasoned developers, but oh, how important it is!
+
+As you develop your new project you may think this isn't that important, since you remember your file names and you can shift easily by searching them in Xcode.
+
+However when some time passes this is no longer the case and finding the file you want to update may take you a while. 
+
+### Organize code specific to your feature
+
+In the root of the project, keep folders with each feature name, some examples we find in most projects are: Authentication, Profile, Home, Navigation, Networking, etc. 
+
+Within these folders, group files tightly related to the feature in question. I like to divide them into different folders as follows: Models, Views, View Models, Managers, and Protocols. 
+
+### Organize common code
+
+In the root, besides the feature-specific folders I like to keep an Extension, Helpers, Common, and Resources folders.
+
+In the Extension folder: keep extensions of the most used native controls and objects. 
+
+In the Helpers folder: keep Constant files and classes you use regularly on the whole codebase. 
+
+In Resources folder: keep any resource file your app will consume, these may be: Property lists, JSONs, Images, etc.
+
+In Common folder: keep classes reused all along the codebase including: views, managers, protocols, as usual separated by type in different folders. 
+
+So the next time I want to do an update, by a glance to the root I know the features my app implements and I can easily navigate to the file I want to update.  
+
+### Files Structure
+
+Within file classes keep related functions together in sections organized with Marks. Some examples for controllers are: View Lifecycle, Layout, User Interaction, Protocol Implementation, etc. 
+
+It is also a nice practice to separate private functions into a private extension, so that a client at a glance at the class can differentiate the interface from the implementation.  
```

Comment:
I don't think it's correct to say at a glance at something. You can say glance at something or at a glance, but both together sounds weird, I might be wrong though. Maybe you could rephrase it like this: (?)

```suggestion
It is also a nice practice to separate private functions into a private extension so that a client can easily differentiate the interface from the implementation by taking a glance at the class.
```